crml/cennzx-spot/src/lib.rs

```diff
@@ -45,8 +62,11 @@ pub trait Trait: system::Trait + generic_asset::Trait {
 	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
 	/// A function type to get an exchange address given the asset ID pair.
 	type ExchangeAddressGenerator: ExchangeAddressFor<Self::AssetId, Self::AccountId>;
-	// TODO: this is just a quick workaround for type conversion issue. should be replaced with proper From/TryFrom Into/TryInfo bounds
-	type AsBalance: From<<Self as generic_asset::Trait>::Balance> + Into<<Self as generic_asset::Trait>::Balance> + As<u128>;
+	// TODO: remove this temp fix for overflow issue when upstream fix ready
+	type BalanceToU128: From<<Self as generic_asset::Trait>::Balance> + Into<u128>;
```

Comment:
Having two types here doesn't seem right. Is there a reason we can't just have one `Balance` type that can convert to/from u128 and generic asset `Balance`. Type hints should be enough to do the correct from/into transformations

Reply:
@shaopengw - can you please help answer this query

Reply:
GA is designed for general purpose, in runtime def the concrete type could vary. Adding `From<u128>` and `To<u128>` bounds would limit the possible concrete types.

Reply:
Oh I see the confusion now. I meant why do we need both `BalanceToU128` and `U128ToBalance` for CENNZ-X? Can we just keep the single `AsBalance` type from before? they both bind to Balance here anyway https://github.com/cennznet/cennznet/pull/38/files#diff-11f21b7f145f98241270cf825e0aaab5R232

Reply:
`AsBalance` is backed by the `As<u128>` to do the conversion, but recently we've been trying to get rid of `As` bounds through runtime modules.

Reply:
Ok, disregarding the name and As trait, shouldn't we just use one bounded type here? Just need to think of the perfect name 😆

Reply:
I tried to use one single bounded type when solving the overflow problem in staking module, but it didn't work. It might be I did something wrongly.😅